### PR TITLE
ConnectToMySQL

### PR DESCRIPTION
**Pull request by Adel-Akbi**
@bastianilso, 
 #9 
After importing the "ConnectToMySQL.unitypakage" to Unity and adding 2  GameObject , I've assigned for  : 

- The first GameObject : named ConnectToMySQL => the scipt "ConnectToMySQL".

- The second GameObject :  Named ConnectToArduino => the script "ConnectToArduino".

The condition saying that, "the user must enter his/her credentials in AAU MySQL Authentification form at runtime and save them to disk (and never ask again)" is respected 💥.

However, the data can be sent and stocked  into the DB (the Arduino must be connected), below, you can see the proof of that:

ProofDB

When @MaximeHollebecq opened the project on his computer, he was connected on the COM12 port (> 10) and me on the COM3 port (<10) which generated an error indicating that the port 12 is not recognized (doesn't exist).

It was therefore necessary to add a syntax on the ConnectoToArduino script to connect to any port, eaither if it's > = 10.

### CODE REVIEW

**Comment by bastianilso** [comment]
@Adel-Akbi awesome - please have @MaximeHollebecq review the code and your commit structure. I will give you a final review afterwards :+1:

**Comment by MaximeHollebecq** [comment]
Great , to be more specific , i suggest you to change the commit message of the commit 
- on the connect to any port , precise that this commit is for any port of the Arduino 
- on the ConnectToMySQL , change the commit message with more explanation of what your code is doing

except these few changes , it seems good to me 👍

File: assets/Scripts/Managers/LoggingManager.cs [hunk 1]
```diff
@@ -155,6 +162,46 @@ public class LoggingManager : MonoBehaviour {
 		date = System.DateTime.Now.ToString("yyyy-MM-dd");
 		time = System.DateTime.Now.ToString("HH:mm:ss:ffff");
 
+	Dictionary <string, List<string>> logs = new Dictionary<string, List<string>>() //create a new dictionary
```
Comment (bastianilso) [inline assets/Scripts/Managers/LoggingManager.cs]: You only need to construct the dictionary once. I suggest you move it to the `Awake()` function. If you need to clear it, you can move it to a separate function which you call, when clearing is needed.

File: assets/Scripts/Managers/LoggingManager.cs [hunk 2]
```diff
@@ -189,7 +236,45 @@ public class LoggingManager : MonoBehaviour {
...
+		logs["DeltaHitX"].Add(_hitPos.x.ToString());
+		logs["DeltaHitY"].Add(_hitPos.y.ToString());
+		logs["TargetDiameter"].Add(_diameter.ToString());
+		logs["ColliderDiameter"].Add(_diameter.ToString());
+		logs["Backtracking"].Add(_backtracking.ToString());
+		logs["ErrorTargetID"].Add(_errorTargetID.ToString());
+		
+
+		connectToMySQL.AddToUploadQueue(logs);
+		connectToMySQL.UploadNow();
```
Comment (bastianilso) [inline assets/Scripts/Managers/LoggingManager.cs]: This code might be called many times and in that case, we are making a lot of upload calls. Here is my suggestion:

- Keep appending new logs to the existing Dictionary, using your `logs[""].Add()` calls.
- Once the test is done, then you call the `connectToMySQL.AddToUploadQueue()` and `connectToMySQL.UploadNow()` functions.

This way, the upload call only happens 1 time, which minimizes traffic. 💾

File: assets/Scripts/Managers/LoggingManager.cs [hunk 3]
```diff
@@ -189,7 +236,45 @@ public class LoggingManager : MonoBehaviour {
 		{
 			writer.WriteLine(currentEntry);
 		}
-	}
+
+		string b = System.Enum.GetName(typeof(GameType), _gameType);         //To Get the name of the enumerator
+
+		logs["Email"].Add("hello@email.test");
```
[the remark this answers is not in the document]
Comment (Adel-Akbi) [reply assets/Scripts/Managers/LoggingManager.cs]: Yes I see. We have to fix it in the futur ... ⚙️

File: assets/Scripts/Managers/LoggingManager.cs (hunk 1 above)
Comment (Adel-Akbi) [reply assets/Scripts/Managers/LoggingManager.cs]: I made some changes by doing what you suggested. So, now the dictionary is moved into the `awak()` function and constructed only once and it cleared with `resetLogs` function 🌖.
I learnt new best practice... thank you 😎

File: assets/Scripts/Managers/LoggingManager.cs (hunk 2 above)
Comment (Adel-Akbi) [reply assets/Scripts/Managers/LoggingManager.cs]: It sounds logical. Now it's done 👍

**Pull request merged**